crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -174,6 +188,27 @@ impl<N, Provider, Pool, Network, Tasks, EvmConfig, BlockExecutor, Consensus>
     RpcModuleBuilder<N, Provider, Pool, Network, Tasks, EvmConfig, BlockExecutor, Consensus>
 where
     N: NodePrimitives,
+    Pool: TransactionPool + Unpin,
+    EvmConfig: ConfigureEvm,
+    Consensus: Clone + Unpin + FullConsensus<N>,
+    Network: Clone,
+    Provider: Clone
+        + DatabaseProviderFactory
+        + ForkChoiceSubscriptions
+        + reth_provider::StaticFileProviderFactory,
+    Tasks: Clone,
+    BlockExecutor: Clone,
+    N: NodePrimitives,
+    Provider: FullRpcProvider<Block = N::Block, Receipt = N::Receipt, Header = N::BlockHeader>
+        + CanonStateSubscriptions<Primitives = N>
+        + AccountReader
+        + ChangeSetReader,
+    Pool: TransactionPool + 'static,
+    Network: NetworkInfo + Peers + Clone + 'static,
+    Tasks: TaskSpawner + Clone + 'static,
+    EvmConfig: ConfigureEvm<Primitives = N>,
+    BlockExecutor: BlockExecutorProvider<Primitives = N>,
+    Network: FullNetwork<Client: BlockClient<Block = <N as NodePrimitives>::Block>>,
```

Comment:
right, if we squash the individual generics in the rpc types then we need to merge them them some in the builder entrypoint 🤔